docs/usage/cmd_deploy.rst

```diff
@@ -0,0 +1,121 @@
+==================================
+Deploying a new application config
+==================================
+
+``shpkpr deploy`` allows you to deploy a new (or changes to an existing) application configuration by rendering and POSTing a JSON template to Marathon::
+
+    $ shpkpr deploy --help
+    Usage: shpkpr deploy [OPTIONS]
+
+      Deploy application from template.
+
+    Options:
+      -t, --template FILENAME  Path of the template to use for deployment.
```

Comment:
The `--marathon_url` is actually documented at the top-level `shpkpr --help` as it's required for most commands. @declantraynor has raised this same point before and I think it makes sense to have it documented with the other options as you suggest.

The problem is the way that click autogenerates help screens. We can override it and change the behaviour to make it more consistent but I don't think it belongs in this PR, I'll address it in a follow-up PR once I get a chance to discuss the approach with @declantraynor